binary/reflect.go

```diff
@@ -803,6 +820,24 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 			WriteTo([]byte("}"), w, n, err)
 		}
 
+	case reflect.Map:
+		WriteTo([]byte("{"), w, n, err)
+		keys := rv.MapKeys()
```

Comment:
Ah right. And I figured we'd never implement c-style because it doesn't help humans and it would break the trees. I think for now I'll close this, use a struct, and enforce order, and we can revisit if there's a more pressing need. Will do tmrw